packages/bus-rdf-update-quads/test/ActorRdfUpdateQuadsDestination-test.ts

```diff
@@ -60,9 +68,92 @@ describe('ActorRdfUpdateQuadsDestination', () => {
       return actor.run({
         quadStreamInsert: new ArrayIterator([]),
         quadStreamDelete: new ArrayIterator([]),
+        context: new ActionContext(),
       }).then(async(output: any) => {
         await expect(output.execute()).resolves.toBeUndefined();
       });
     });
   });
+
+  describe('An ActorRdfUpdateQuadsDestination instance with rdfjs source', () => {
+    const actor = new (<any> ActorRdfUpdateQuadsDestination)({ name: 'actor', bus });
+    actor.getDestination = context => {
+      // Console.log('get destination called')
+      return Promise.resolve(new RdfJsQuadDestination(context.get(KeysRdfUpdateQuads.destination)));
+    };
+
+    it('should deskolemize quads retrieved from the same source', async() => {
+      const q = quad(blankNode('i'), namedNode('http://example.org#type'), namedNode('http://example.org#thing'));
+
+      const store = new Store();
+      const context: IActionContext = new ActionContext({
+        [KeysRdfUpdateQuads.destination.name]: store,
+        [KeysRdfResolveQuadPattern.sourceIds.name]: new Map([[ store, '1' ]]),
+      });
+
+      const output: IActorRdfUpdateQuadsOutput = await actor.run({
+        quadStreamInsert: new ArrayIterator([
+          FederatedQuadSource.skolemizeQuad(q, '1'),
+        ], { autoStart: false }),
+        quadStreamDelete: new ArrayIterator([], { autoStart: false }),
+        context,
+      });
+
+      await output.execute();
+
+      expect(store.getQuads(null, null, null, null)).toEqual([ q ]);
+    });
+
+    it('should not deskolemize quads retrieved from a different source', async() => {
+      const q = quad(blankNode('i'), namedNode('http://example.org#type'), namedNode('http://example.org#thing'));
+      const skolemized = quad(
+        blankNode('bc_1_i'), namedNode('http://example.org#type'), namedNode('http://example.org#thing'),
+      );
+
+      const store = new Store();
+      const context: IActionContext = new ActionContext({
+        [KeysRdfUpdateQuads.destination.name]: store,
+        [KeysRdfResolveQuadPattern.sourceIds.name]: new Map([[ store, '2' ]]),
+      });
+
+      const output: IActorRdfUpdateQuadsOutput = await actor.run({
+        quadStreamInsert: new ArrayIterator([
+          FederatedQuadSource.skolemizeQuad(q, '1'),
+        ], { autoStart: false }),
+        quadStreamDelete: new ArrayIterator([], { autoStart: false }),
+        context,
+      });
+
+      await output.execute();
+
+      expect(store.getQuads(null, null, null, null)).toEqual([ skolemized ]);
+    });
+  });
 });
+
+/**
+ * A quad destination that wraps around an {@link RDF.Store}.
+ */
+class RdfJsQuadDestination {
+  private readonly store;
+
+  public constructor(store: RDF.Store) {
+    this.store = store;
+  }
+
+  protected promisifyEventEmitter(eventEmitter: EventEmitter): Promise<void> {
```

Comment:
I often use this pattern myself as well, we should probably create a dedicated helper (package?) for this at some point. (but not in this PR)